HLTrigger/Muon/interface/HLTTriMuonIsolation.h

```diff
@@ -0,0 +1,261 @@
+                    int                      charge   = Daughters[0].charge() + Daughters[1].charge() + Daughters[2].charge();
+                    math::XYZTLorentzVectorD taup4    = Daughters[0].p4()     + Daughters[1].p4()     + Daughters[2].p4()    ;
+                    int                      tauPdgId = charge > 0? 15 : -15;
+
+                    Tau.setP4(taup4);
+                    Tau.setCharge(charge);
+                    Tau.setPdgId(tauPdgId);
+                    Tau.setVertex((Daughters)[0].vertex()); // assign the leading muon vertex as tau vertex
+
+                    // the three muons must be close to each other in Z
+                    if (std::abs(Tau.daughter(0)->vz() - Tau.vz()) > MaxDZ_) continue;
+                    if (std::abs(Tau.daughter(1)->vz() - Tau.vz()) > MaxDZ_) continue;
+                    if (std::abs(Tau.daughter(2)->vz() - Tau.vz()) > MaxDZ_) continue;
+                    
+                    // require muons to be collimated
+                    bool collimated = true;           
+                    for (auto const &idau : Daughters){
+                        if (reco::deltaR2(Tau.p4(), idau.p4()) > MaxTriMuonRadius_*MaxTriMuonRadius_) {
+                            collimated = false;
+                            break;
+                        }
+                    }
+                                  
+                    if (!collimated) continue;
+                        
+                    // a good tau, at last                      
+                    Taus->push_back(Tau);
+                }
+            }
+        }
+
+        // Loop over taus and further select
+        for (reco::CompositeCandidateCollection::const_iterator itau = Taus->begin(); itau != Taus->end(); ++itau){
+            if (         itau->pt()   < TriMuonPtCut_  ) continue;
+            if (         itau->mass() < MinTriMuonMass_) continue;
+            if (         itau->mass() > MaxTriMuonMass_) continue;
+            if (std::abs(itau->eta()) > TriMuonEtaCut_ ) continue;
+            if (itau->daughter(0)->pt() < Muon1PtCut_  ) continue;
+            if (itau->daughter(1)->pt() < Muon2PtCut_  ) continue;
+            if (itau->daughter(2)->pt() < Muon3PtCut_  ) continue;
+            if ((std::abs(itau->charge()) != TriMuonAbsCharge_) & (TriMuonAbsCharge_ >= 0)) continue;
+            
+            // remove the candidate pt from the iso sum
+            double sumPt = -itau->pt(); 
+            
+            // compute iso sum pT
+            for (reco::TrackCollection::const_iterator itrk = IsoTracks->begin(); itrk != IsoTracks->end(); ++itrk){
+                if (reco::deltaR2(itrk->momentum(), itau->p4()) > IsoConeSize_*IsoConeSize_) continue;
+                if (std::abs(itrk->vz() - itau->vz()) > MaxDZ_) continue;
+                sumPt += itrk->pt();
+            }
+            
+            // apply the isolation cut
+            if ((std::max(0., sumPt) > (EnableAbsIso_ * ChargedAbsIsoCut_)) || 
```

Comment:
in practice it is not needed.

I think I put it because I didn't like the idea of a negative sum pt (whilst it might happen because tracks from different tracking sequences are used at the same time).

Will remove it altogether.